### DISCUSSION TITLE
Undefined Reference to Assimp::Importer, but using C headers compiles fine.

### DISCUSSION THREAD

**Discussion by KimeraRoyal** [assimp/assimp#5165] [Q&A]
I've encountered an interesting issue when using Assimp (built with vcpkg) and CMake, on CLion. Building using the C header (assimp/cimport.h) I encounter no issues, but if I try to include assimp/Importer.hpp and create / use an instance of Assimp::Importer, I get an error thrown upon linking the related source file that says "undefined reference to `Assimp::Importer::Importer()'".

I'm curious what the issue could be, it's odd that the C API works completely fine as I would have assumed it's simply me forgetting to link the library. Has anybody run into this issue before, and do you know any potential fix? Sorry if somebody has brought this up before - I did do my best to search for this issue but it only seems to come up with people who had issues linking the library in its entirety.

**Comment by AdamRSawyer** [comment]
I am having the same issue, I have not been able to find a fix. I know it's been over a year but thank you for pointing out that you can use the C header instead!

**Comment by TeMaToSer** [comment]
I had the same error when tried to use x64 lib version instead of x86. So it could be some really bug to fix.
With x86 setup everything fine.

Thank you for your solution, it is really working. Another proof that C is elegant power!
